crates/meilisearch/tests/search/facet_search.rs

```diff
@@ -436,3 +494,124 @@ async fn deactivate_facet_search_add_documents_and_reset_facet_search() {
     assert_eq!(code, 200, "{}", response);
     assert_eq!(dbg!(response)["facetHits"].as_array().unwrap().len(), 2);
 }
+
+#[actix_rt::test]
+async fn facet_search_with_filterable_attributes_rules() {
+    test_settings_documents_indexing_swapping_and_facet_search(
+        &DOCUMENTS,
+        &json!({"filterableAttributes": ["genres"]}),
+        &json!({"facetName": "genres", "facetQuery": "a"}),
+        |response, code| {
+            snapshot!(code, @"200 OK");
+            snapshot!(response["facetHits"], @r###"[{"value":"Action","count":3},{"value":"Adventure","count":2}]"###);
+        },
+    )
+    .await;
+
+    test_settings_documents_indexing_swapping_and_facet_search(
+        &DOCUMENTS,
+        &json!({"filterableAttributes": [{"patterns": ["genres"], "features": {"facetSearch": true, "filter": {"equality": false, "comparison": false}}}]}),
+        &json!({"facetName": "genres", "facetQuery": "a"}),
+        |response, code| {
+            snapshot!(code, @"200 OK");
+            snapshot!(response["facetHits"], @r###"[{"value":"Action","count":3},{"value":"Adventure","count":2}]"###);
+        },
+    ).await;
+
+    test_settings_documents_indexing_swapping_and_facet_search(
+        &NESTED_DOCUMENTS,
+        &json!({"filterableAttributes": ["doggos.name"]}),
+        &json!({"facetName": "doggos.name", "facetQuery": "b"}),
+        |response, code| {
+            snapshot!(code, @"200 OK");
+            snapshot!(response["facetHits"], @r###"[{"value":"bobby","count":1},{"value":"buddy","count":1}]"###);
+        },
+    )
+    .await;
+
+    test_settings_documents_indexing_swapping_and_facet_search(
+        &NESTED_DOCUMENTS,
+        &json!({"filterableAttributes": [{"patterns": ["doggos.name"], "features": {"facetSearch": true, "filter": {"equality": false, "comparison": false}}}]}),
+        &json!({"facetName": "doggos.name", "facetQuery": "b"}),
+        |response, code| {
+            snapshot!(code, @"200 OK");
+            snapshot!(response["facetHits"], @r###"[{"value":"bobby","count":1},{"value":"buddy","count":1}]"###);
+        },
+    ).await;
+}
+
+#[actix_rt::test]
+async fn facet_search_with_filterable_attributes_rules_errors() {
+    test_settings_documents_indexing_swapping_and_facet_search(
+        &DOCUMENTS,
+        &json!({"filterableAttributes": ["genres"]}),
+        &json!({"facetName": "invalid", "facetQuery": "a"}),
+        |response, code| {
+            snapshot!(code, @"400 Bad Request");
+            snapshot!(response["message"], @r###""Attribute `invalid` is not facet-searchable. Available facet-searchable attributes are: `genres`. To make it facet-searchable add it to the `filterableAttributes` index settings.""###);
+        },
+    )
+    .await;
+
+    test_settings_documents_indexing_swapping_and_facet_search(
+      &DOCUMENTS,
+      &json!({"filterableAttributes": [{"patterns": ["genres"]}]}),
```

Comment:
This error is a bit confusing because `genres` **is** already in the list of patterns of the `filterableAttributes` index settings. I think you should specify that `features.facetSearch` should be set to `true` as well.

The error message could become:

```md
Attribute `genres` is not facet-searchable.
  - note: this index does not have a filterable attribute pattern that is facet searchable.
  - hint: add `{ "patterns": ["genres"], "features": { "facetSearch":true } }` to `filterableAttributes`
```

Reply:
Referenced and detailed as https://github.com/meilisearch/meilisearch/issues/5411.

Resolving for now